raiden/utils/upgrades.py

```diff
@@ -109,93 +126,78 @@ class UpgradeManager:
 
     Upgrade procedure:
 
+    - Delete corrupted databases.
     - Copy the old file to the latest version (e.g. copy version v16 as v18).
     - In a transaction: Run every migration. Each migration must decide whether
       to proceed or not.
-    - If a single migration fails: The transaction is not commited and the DB
-      copy is deleted.
-    - If every migration succeeds: Rename the old DB.
     """
 
     def __init__(self, db_filename: str, web3: Web3 = None):
-        base_name = os.path.basename(db_filename)
-        match = VERSION_RE.match(base_name)
-        assert match, f'Database name "{base_name}" does not match our format'
-
         self._current_db_filename = Path(db_filename)
         self._web3 = web3
-        self._current_version = match.group(1)
+        self._current_version = get_file_version(db_filename)
 
     def run(self):
-        """
-        The `_current_db_filename` is going to hold the filename of the database
-        with the new version. However, the previous version's data
-        is going to exist in a file whose name contains the old version.
-        Therefore, running the migration means that we have to copy
-        all data to the current version's database and execute the migration
-        functions.
-        """
+        # First clear up any partially upgraded databases.
+        #
+        # A database will be partially upgraded if the process receives a
+        # SIGKILL/SIGINT while executing migrations. NOTE: It's very probable
+        # the content of the database remains consistent, because the upgrades
+        # are executed inside a migration, however making a second copy of the
+        # database does no harm.
         paths = glob(f'{self._current_db_filename.parent}/v*_log.db')
-        older_file = older_db_file(paths)
-
-        if older_file is None or older_file == str(self._current_db_filename):
-            return
-
-        old_db_filename = Path(older_file)
-
-        with get_file_lock(old_db_filename), get_file_lock(self._current_db_filename):
-            if self._current_db_filename.exists():
-                db_version = get_db_version(self._current_db_filename)
-
-                # The current version matches our target version, nothing to do.
-                if db_version == RAIDEN_DB_VERSION:
-                    return
+        valid_db_names = filter_db_names(paths)
+        delete_dbs_with_failed_migrations(valid_db_names)
 
-                if db_version > RAIDEN_DB_VERSION:
-                    raise RuntimeError(
-                        f'Database version higher then expected. '
-                        f'It is {db_version} should be {self._current_version}',
-                    )
+        # At this point we know every file version and db version match
+        # (assuming there are no concurrent runs).
+        paths = glob(f'{self._current_db_filename.parent}/v*_log.db')
```

Comment:
We must get a new list of files from the FS because a corrupt database could have been deleted.